File: TraderCore/TradingCosts.py
class TradingCosts(object):
   def __init__(self, slipage, commission_per_share, sec_transaction, finra_activity_fee, min_fee):
      self.slipage = slipage
      self.commission = commission_per_share
      self.sec_transaction = sec_transaction
      self.finra_activity = finra_activity_fee
      self.min_fee = min_fee

   def compute_cost(self, quantity, price, buy):
      share_cost = max((self.commission + self.finra_activity) * quantity, self.min_fee)
      total_slipage = self.slipage * quantity
      if not buy:
         sec_fee = (price * quantity + total_slipage) * self.sec_transaction
      else:
         sec_fee = 0.0
      return share_cost + sec_fee, total_slipage

   def __str__(self) -> str:
      return f"Costs: {self.slipage}, {self.commission}, {self.sec_transaction}, {self.finra_activity}, {self.min_fee}"

ibkr_trading_costs = TradingCosts(0.02, 0.005, 0.0000229, 0.00013, 1.00)

File: lightning/TradingCosts.cpp
#include "TradingCosts.h"
#include <algorithm>

TradingCosts::TradingCosts(double slipage, double commission_per_share, double sec_transaction, double finra_activity_fee, double min_fee) :
   _slipage(slipage),
   _commission(commission_per_share),
   _sec_transaction(sec_transaction),
   _finra_activity(finra_activity_fee),
   _min_fee(min_fee)
{
   return;
}

double TradingCosts::compute_cost(int quantity, double price, bool buy)
   {
      auto share_cost = std::max((_commission + _finra_activity) * quantity, _min_fee);
      auto total_slipage = _slipage * quantity;
      double sec_fee = 0.0;
      if (!buy)
      {
         sec_fee = (price * quantity + total_slipage) * _sec_transaction;
      }
      else
      {
         sec_fee = 0.0;
      }
      return share_cost + sec_fee, total_slipage;
   }